assets/app/scripts/services/docker.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("DockerImageGeneratorService", function($filter, ApplicationGenerator, DataService) {
+    var findImage = function(name, projectContext) {
+      var importImage = {
+        kind: "ImageStreamImport",
+        apiVersion: "v1",
+        metadata: {
+          name: "newapp",
```

Comment:
I'm not sure. FWIW, it's what the CLI does:

https://github.com/spadgett/origin/blob/ecf1e9d70dfa964dff684d008bb44991cb39662f/pkg/generate/app/dockerimagelookup.go#L195